### ISSUE TITLE
federation tester incorrectly says it times out when .well-known has a time-out, but an SRV record is present

### ISSUE THREAD

**Issue by mvgorcum** [matrix-org/matrix-federation-tester#113]
**Describe the bug**

**To Reproduce**
Run federation tester on a server that has SRV domain delegation, and a time-out on https://server_name/.well-known/matrix/server

**Expected behavior**
I expected the federation tester to tell me the server works correctly, and optionally to mention a time-out on the .well-known lookup.

The tester gives a time-out error, but it doesn't clearly state that .well-known has a time-out:

> Got 1 connection report.

>There was an error looking up homeserver version information: Get matrix://balsillie.net/_matrix/federation/v1/version: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

I would say it's technically correct, but hard to interpret.
**Screenshots**
![image](https://user-images.githubusercontent.com/24435423/104813245-39f0d800-5808-11eb-9e30-da927efb4a15.png)

**Comment by Bubu** [comment]
The json report correctly mentions that the well-known request timed out (in addition to the confusing general matrix:// timeout report). But this is apprently not surfaced to the frontend:

![image](https://user-images.githubusercontent.com/105185/104813367-006c9c80-5809-11eb-834b-8b8c1392448a.png)

**Comment by aaronraimist** [comment]
This repo is just the federation tester itself, which provides the JSON output. The federationtester.matrix.org UI is on https://github.com/matrix-org/fed-tester-ui

**Comment by Bubu** [comment]
Right, this is a separate frontend issue as well for not reporting the well known error at all. But the json output produces the misleading timeout on federation request in the first place. (The timeout only happens on the well-known).

**Comment by strifel** [comment]
I feel like this should still be fixed, as having only the srv works ok as well and the tester produces wrong output in that case

**Comment by nrgill28** [comment]
Same issue happens on my homeserver. I also have an SRV record setup pointing to my matrix subdomain on port 8448, but the federation tester seems to be trying to make this request over port 443 which I don't have open and it causes a timeout.

https://federationtester.matrix.org/#hobscrk.space
https://matrix.hobscrk.space:8448/_matrix/federation/v1/version
